parquet-column/src/main/java/org/apache/parquet/column/statistics/SizeStatistics.java

```diff
@@ -0,0 +1,254 @@
+   * physical type.
+   */
+  private long unencodedByteArrayDataBytes;
+  /**
+   * When present, there is expected to be one element corresponding to each
+   * repetition (i.e. size=max repetition_level+1) where each element
+   * represents the number of times the repetition level was observed in the
+   * data.
+   *
+   * This field may be omitted if max_repetition_level is 0 without loss
+   * of information.
+   */
+  private List<Long> repetitionLevelHistogram;
+  /**
+   * Same as repetition_level_histogram except for definition levels.
+   *
+   * This field may be omitted if max_definition_level is 0 or 1 without
+   * loss of information.
+   */
+  private List<Long> definitionLevelHistogram;
+
+  /**
+   * Whether the statistics has valid value.
+   *
+   * It is true by default. Only set to false while it fails to merge statistics.
+   */
+  private boolean hasValue = true;
+
+  /**
+   * Builder to create a SizeStatistics.
+   */
+  public static class Builder {
+    private PrimitiveType type;
+    private long unencodedByteArrayDataBytes;
+    private long[] repetitionLevelHistogram;
+    private long[] definitionLevelHistogram;
+
+    /**
+     * Create a builder to create a SizeStatistics.
+     *
+     * @param type physical type of the column associated with this statistics
+     * @param maxRepetitionLevel maximum repetition level of the column
+     * @param maxDefinitionLevel maximum definition level of the column
+     */
+    public Builder(PrimitiveType type, int maxRepetitionLevel, int maxDefinitionLevel) {
+      this.type = type;
+      this.unencodedByteArrayDataBytes = 0L;
+      repetitionLevelHistogram = new long[maxRepetitionLevel + 1];
+      definitionLevelHistogram = new long[maxDefinitionLevel + 1];
+      Arrays.fill(repetitionLevelHistogram, 0L);
+      Arrays.fill(definitionLevelHistogram, 0L);
+    }
+
+    /**
+     * Add repetition and definition level of a value to the statistics.
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
+     *
+     * @param repetitionLevel repetition level of the value
+     * @param definitionLevel definition level of the value
+     */
+    public void add(int repetitionLevel, int definitionLevel) {
+      Preconditions.checkArgument(0 <= repetitionLevel && repetitionLevel < repetitionLevelHistogram.length,
+        "repetitionLevel %s is out of range [0, %s]", repetitionLevel, repetitionLevelHistogram.length - 1);
+      Preconditions.checkArgument(definitionLevel < definitionLevelHistogram.length,
+        "definitionLevel %s is out of range [0, %s]", definitionLevel, definitionLevelHistogram.length - 1);
+      repetitionLevelHistogram[repetitionLevel]++;
+      definitionLevelHistogram[definitionLevel]++;
+    }
+
+    /**
+     * Add repetition and definition level of a value to the statistics.
+     * It is called when value is null, or the column is not of BYTE_ARRAY type.
```

Comment:
I think here the column could be `BYTE_ARRAY`. The `BYTE_ARRAY` is `Binary`, right?

Reply:
The check here simply ignores the Binary value from FIXED_SIZE_BYTE_ARRAY type which we do not care.